tools/importer-rest-api-specs/components/parser/parser.go

```diff
@@ -66,6 +66,34 @@ func (d *SwaggerDefinition) parse(serviceName, apiVersion string, resourceProvid
 		resourcesOut[resourceName] = updated
 	}
 
+	// TODO this can be removed when https://github.com/hashicorp/pandora/issues/3725 is resolved
+	if !strings.EqualFold(serviceName, "datafactory") {
+		// discriminator implementations that are defined in separate files with no link to a swagger tag
+		// are not parsed. So far there are two known instances of this (Data Factory, Chaos Studio) where the
+		// files are defined in a nested directory e.g. d.Name = /Types/Capabilities
+		swaggerFileName := strings.Split(d.Name, "/")
+		if len(resources) == 0 && len(swaggerFileName) > 2 {
+			// if we're here then there is no tag in this file, so we'll use the file name
+			inferredTag := cleanup.PluraliseName(swaggerFileName[len(swaggerFileName)-1])
+			normalizedTag := cleanup.NormalizeResourceName(inferredTag)
+
+			result, err := d.findOrphanedDiscriminatedModels()
+			if err != nil {
+				return nil, fmt.Errorf("finding orphaned discriminated models in %q: %+v", d.Name, err)
+			}
+
+			// this is to avoid the creation of empty packages/directories in the api definitions
+			if len(result.Models) > 0 || len(result.Constants) > 0 {
+				resource := models.AzureApiResource{
+					Constants: result.Constants,
+					Models:    result.Models,
+				}
+				resource = normalizeAzureApiResource(resource)
```

Comment:
Although I'm not in agreement that it's necessary I've added the logic to merge the resource information for a tag if it already exists since there isn't any harm in adding it.

Because this function is called on a per file basis, if any tags or operations were defined in the file it would be pulled out either here
https://github.com/hashicorp/pandora/blob/793c3e5a535cf5d860f94ab3a3413412ac35c5f0/tools/importer-rest-api-specs/components/parser/parser.go#L24
or here
https://github.com/hashicorp/pandora/blob/793c3e5a535cf5d860f94ab3a3413412ac35c5f0/tools/importer-rest-api-specs/components/parser/parser.go#L37-L39
resulting in a populated `resources` variable which means we would never enter this part of the function and no prior tags could exist if we ended up here.